kmk/extensions/lock_status.py

```diff
@@ -49,17 +58,28 @@ def on_powersave_enable(self, sandbox):
     def on_powersave_disable(self, sandbox):
         return
 
+    @property
+    def report_updated(self):
+        return self._report_updated
+
+    def check_state(self, lock_code):
+        if self.report:
```

Comment:
This tripped me up. I think we could add a comment "This is false if there's no valid report, or all report bits are zero".

Reply:
Hmm, valid point here... Should we change this to `if self.report is not None`? While functionally it would return the same value, it is more correct.

Reply:
In my opinion a comment is sufficient.
I don't mind the implicit double check, and in case of zero it's a bit faster then checking against zero again later.